packages/js-lib/src/json-schema/jsonSchemaBuilder.ts

```diff
@@ -58,9 +61,38 @@ export const j = {
       $ref,
     })
   },
-  enum<T = unknown>(enumValues: T[]) {
-    return new JsonSchemaAnyBuilder<T, JsonSchemaEnum<T>>({ enum: enumValues })
+
+  enum<const T extends readonly (string | number | boolean | null)[] | StringEnum | NumberEnum>(
+    values: T,
+  ) {
+    let enumValues: readonly (string | number | boolean | null)[] | undefined
+
+    if (Array.isArray(values)) {
+      enumValues = values
+    } else if (typeof values === 'object') {
+      if (isNumberEnum(values)) {
```

Comment:
Performance thinking.
You're going over Object.values of the enum twice.

What if we had `getEnumType(input): 'number' | 'string' | undefined` that would do 1 pass and "detect enum type" (with an old-man's yelling for-loop). Undefined would mean "it's neither"